### PR TITLE
feat(errors): add error codes to error context

### LINKED ISSUE #36: Design error system for CLI
We need to figure out what the cross-package error signaling, code, messaging, etc, system is going to be for `ds`. Do we use some kind of centralized error code system? Do we stick with `anyhow`?

### PR DESCRIPTION
**Pull request by zkat**
Fixes: https://github.com/entropic-dev/ds/issues/36

### CODE REVIEW

File: commands/cmd-ping/src/lib.rs [hunk 1]
```diff
@@ -73,7 +74,7 @@ impl PingCmd {
         if self.json {
             let details: Value =
                 serde_json::from_str(&res.body_string().await.unwrap_or("{}".into()))
-                    .context("Failed to parse response body")?;
+                    .context(Ctx::DS1011)?;
```
Comment (chrisdickinson) [inline commands/cmd-ping/src/lib.rs:90]: I love the usage!

File: packages/ds-error-context/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,57 @@
+use std::path::PathBuf;
+
+use derive_more::Display;
+
+#[derive(Display)]
```
Comment (chrisdickinson) [inline packages/ds-error-context/src/lib.rs:9]: Oh interesting! I had been planning on checking out [`thiserror`](https://crates.io/crates/thiserror) which seems to have a similar API, but I guess we won't need that if we lean towards attaching `context()` to errors instead?

**Review by chrisdickinson — APPROVED** [review]
This is great! I have to admit I'm still a little foggy on the boundaries between `.context()` and full-blown error types like `thiserror`; in any case this is a smart time to start codifying client errors.

File: packages/ds-error-context/src/lib.rs (hunk 2 above)
Comment (zkat) [reply packages/ds-error-context/src/lib.rs:9]: So... we kinda use both. `thiserror` is used for programmatic errors that people can check for, handle, etc. The thing about this "error" enum is that it's really just an enum of strings that we can use to add context for end-users to reference. There can be an M:N behavior between actual errors and error contexts, for example. They're distinct entities.

**Pull request merged**